code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -155,9 +159,95 @@
 
 	succeed_activate()
 	add_cooldown()
-	
+
 	new /obj/effect/alien/egg/gas(A.loc)
 	qdel(alien_egg)
 
 	GLOB.round_statistics.defiler_inject_egg_neurogas++
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "defiler_inject_egg_neurogas")
+
+// ***************************************
+// *********** Reagent selection
+// ***************************************
+/datum/action/xeno_action/select_reagent
+	name = "Select Reagent"
+	action_icon_state = "select_reagent0"
+	mechanics_text = "Selects which reagent to inject. Hemodile slows by 25%, increased to 50% with neurotoxin present, and deals 20% of dmage received as stamina damage. Transvitox after ~4s converts brute/burn damage to toxin based on 40% of damage received up to 45 toxin on target."
+	use_state_flags = XACT_USE_BUSY
+
+/datum/action/xeno_action/select_reagent/update_button_icon()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/atom/A = X.selected_reagent
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/actions.dmi', button, initial(A.name))
+	return ..()
+
+/datum/action/xeno_action/select_reagent/action_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/list/available_reagents = X.xeno_caste.available_reagents_define
+	var/i = available_reagents.Find(X.selected_reagent)
+	if(length(available_reagents) == i)
+		X.selected_reagent = available_reagents[1]
+	else
+		X.selected_reagent = available_reagents[i+1]
+	var/atom/A = X.selected_reagent
+	to_chat(X, "<span class='notice'>We will now slash with <b>[initial(A.name)]</b>.</span>")
+	update_button_icon()
+	return succeed_activate()
+
+// ***************************************
+// *********** Reagent slash
+// ***************************************
+/datum/action/xeno_action/activable/reagent_slash
+	name = "Reagent Slash"
+	mechanics_text = "Deals damage 4 times and injects 3u of selected reagent per slash. Can move while slashing."
+	ability_name = "reagent slash"
+	cooldown_timer = 6 SECONDS
+	plasma_cost = 100
+	target_flags = XABB_MOB_TARGET
+
+/datum/action/xeno_action/activable/reagent_slash/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(get_dist(owner, A) > 1)
+		to_chat(owner, "<span class='warning'>We need to be next to our prey.</span>")
+		return FALSE
+	if(!A?.can_sting())
+		to_chat(owner, "<span class='warning'>Our slash won't affect this target!</span>")
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/reagent_slash/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/slash_count = 4
+	var/reagent_transfer_amount = 3
```

Comment:
#define in the ability defines not variables

Reply:
changed